projects/core/koin-core/src/commonMain/kotlin/org/koin/core/module/Module.kt

```diff
@@ -235,12 +235,19 @@ operator fun List<Module>.plus(module: Module): List<Module> = this + listOf(mod
  * Run through the module list to flatten all modules & submodules
  */
 @OptIn(KoinInternalApi::class)
-fun flatten(modules: List<Module>): Set<Module> {
-    fun flat(modules: List<Module>, newModules: MutableSet<Module>){
-        modules.forEach{
-            newModules += it
-            flat(it.includedModules,newModules)
+fun flatten(modules: List<Module>): Set<Module> {    
+    val flatten = HashSet<Module>()
+    val stack = ArrayDeque<Module>(modules)
+
+    while (stack.isNotEmpty()) {
+        val current = stack.removeLast()
+        flatten.add(current)
+        current.includedModules.forEach {
```

Comment:
Please use a `for` loop instead, `forEach` is designed to be used in "chains" or if the receiver is nullable.

For a reference, please see the [Kotlin Style Guide](https://kotlinlang.org/docs/coding-conventions.html#loops):
> `forEach` (prefer using a regular for loop instead, unless the receiver of `forEach` is nullable or `forEach` is used as part of a longer call chain.